src/backend/executor/nodeAgg.c

```diff
@@ -1950,6 +1950,21 @@ ExecInitAgg(Agg *node, EState *estate, int eflags)
 	 * initialize child nodes
 	 */
 	outerPlan = outerPlan(node);
+	if (outerPlan->type == T_ExternalScan) {
+		/*
+		 * Hack to indicate to PXF when there is an external scan
+		 */
+		if (aggstate->aggs && list_length(aggstate->aggs) == 1) {
+			foreach(l, aggstate->aggs) {
```

Comment:
if the size of `aggstate->aggs` is already 1, do we need use `foreach` to iterate?